src/components/common/controllers/drag.ts

```diff
@@ -267,16 +313,30 @@ class DragController implements ReactiveController {
     this._updatePosition(event);
     this._updateMatcher(event);
 
-    const parameters = { event, state: this._stateParameters };
+    this._direction = this._trackPointerMovement(event.clientX, event.clientY);
+
+    const parameters = {
+      event,
+      state: this._stateParameters,
+      direction: this._direction,
+    };
```

Comment:
If `direction` is a property of the `DragState` you can shove those calculations in the `_stateParameters` getter and only update the previous pointer coordinates here